From 655aa59eb86dc7a458f0b3e49f1d1186b728e043 Mon Sep 17 00:00:00 2001
Message-ID: <655aa59eb86dc7a458f0b3e49f1d1186b728e043.1719691458.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1719691458.git.sharlatanus@gmail.com>
References: <cover.1719691458.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Fri, 7 Jun 2024 22:33:22 +0100
Subject: [PATCH 13/26] gnu: Add go-atomicgo-dev-assert.

* gnu/packages/golang-check.scm (go-atomicgo-dev-assert): New variable.

Change-Id: I987bdc50978da3cb6758d99bb4330e0a3bf3d14a
---
 gnu/packages/golang-check.scm | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/gnu/packages/golang-check.scm b/gnu/packages/golang-check.scm
index dd0b71ff13..00ea185cbc 100644
--- a/gnu/packages/golang-check.scm
+++ b/gnu/packages/golang-check.scm
@@ -62,6 +62,33 @@ (define-module (gnu packages golang-check)
 ;;; Libraries:
 ;;;
 
+(define-public go-atomicgo-dev-assert
+  (package
+    (name "go-atomicgo-dev-assert")
+    (version "0.0.2")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/atomicgo/assert")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "1ra5bx3w6vynwbxgsz5knibk2xwmfi6654fsi29zsmk77f39g8vv"))))
+    (build-system go-build-system)
+    (arguments
+     (list
+      #:go go-1.21
+      #:import-path "atomicgo.dev/assert"))
+    (home-page "https://atomicgo.dev/assert")
+    (synopsis "Go package with tons of assertions")
+    (description
+     "Package assert provides obj set of assertion functions.  Every assertion
+function returns obj boolean.  This package does not integrate into the
+testing package automatically and requires to check the returning boolean
+value and call @code{t.Fatal()} if the assertion fails.")
+    (license license:expat)))
+
 (define-public go-github-com-alecthomas-assert-v2
   (package
     (name "go-github-com-alecthomas-assert-v2")
-- 
2.41.0

